--- pysmbc-1.0.13/context.c.orig	2012-02-15 13:00:15.000000000 +0100
+++ pysmbc-1.0.13/context.c	2012-07-22 18:21:16.919596736 +0200
@@ -78,6 +78,7 @@
 			 &use_username,
 			 &use_password))
     {
+      Py_DECREF (result);
       debugprintf ("<- auth_fn(), incorrect callback result\n");
       return;
     }
@@ -85,6 +86,7 @@
   strncpy (workgroup, use_workgroup, wgmaxlen);
   strncpy (username, use_username, unmaxlen);
   strncpy (password, use_password, pwmaxlen);
+  Py_DECREF (result);
   debugprintf ("<- auth_fn(), got callback result\n");
 }
 
